atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTimedOutException.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * (c) Copyright 2024 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import com.palantir.common.exception.AtlasDbDependencyException;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.exceptions.SafeExceptions;
+
+// Added the below suppression because the Runtime class compileTime string check sees the + in the static final as
+// non compile time safe. But no other way to achieve as we are using a java version that does not have text blocks.
+@SuppressWarnings("CompileTimeConstant")
+public class CassandraTimedOutException extends AtlasDbDependencyException {
+    private static final long serialVersionUID = 1L;
+
+    private static final String LOG_MESSAGE =
+            "Cassandra query threw a TimedOut exception. Possible reasons and actions to resolve include:\n"
+                + "1. Reason: AtlasDB clients are requesting too much data from Cassandra.\n"
+                + "   Resolution: Change the query to request less data.\n"
+                + "2. Reason: Data that has been deleted is being read in the query (e.g. A large amount of"
+                + " tombstones).\n"
+                + "   Resolution: Run a compaction on your Cassandra server.\n"
```

Comment:
For this one there's probably something around checking sweep (and then compaction).

Reply:
Changed